### PR TITLE
This PR addresses #14, #28 and #36

### PR DESCRIPTION
**Pull request by ibaldin**
Still WiP - initial pull

### CODE REVIEW

**Comment by ibaldin** [comment]
Requesting review at this point - I removed one of the tickets I was going to address in this PR because it is already huge. This requires an update to the control framework code that looks at delegations. Delegations now have a single unified format:
- Pool definition
```
{ <id>: {
  ‘pool_id’: <pool id string>,
  [‘capacities’ | 'labels']: <capacity or label dictionary> } 
```
<id> can be delegation id (e.g. 'default' or 'primary') on ARM/ADM graphs or it can be graph ID (ADM graph id) on CBM graphs. <pool id string> is any string except '_' (see below).
- Pool reference
```
{ <id>: {
  'pool': <pool id string> {
```
- Single element pool (a pool that covers one and only one node)
```
{ <id>: {
  'pool_id': '_',
  ['capacities' | 'labels']: <capacity or label dictionary> }
```
I will provide the CF code update as well as a separate PR.

File: fim/graph/abc_property_graph_constants.py [hunk 1]
```diff
@@ -58,10 +59,13 @@ class ABCPropertyGraphConstants(ABC):
...
     PROP_RESERVATION_INFO = 'ReservationInfo'
     PROP_NODE_MAP = "NodeMap"
+    PROP_STRUCTURAL_INFO = "StructuralInfo"
+    PROP_STITCH_NODE = "StitchNode"
     # these properties get validated to be valid JSON objects whenever someone validates the graph
     JSON_PROPERTY_NAMES = [PROP_LABELS, PROP_CAPACITIES, PROP_LABEL_DELEGATIONS,
                            PROP_CAPACITY_DELEGATIONS, PROP_LABEL_ALLOCATIONS,
-                           PROP_CAPACITY_ALLOCATIONS, PROP_RESERVATION_INFO]
+                           PROP_CAPACITY_ALLOCATIONS, PROP_RESERVATION_INFO,
+                           PROP_RESERVATION_INFO, PROP_STRUCTURAL_INFO]
```
Comment (kthare10) [inline fim/graph/abc_property_graph_constants.py:68]: PROP_RESERVATION_INFO is duplicated here.

Comment (ibaldin) [reply fim/graph/abc_property_graph_constants.py]: Added a fix

**Review by kthare10 — APPROVED** [review]
Looks good!

**Pull request merged**